torch/nn/modules/container.py

```diff
@@ -186,25 +218,47 @@ def __iter__(self):
     def __iadd__(self, parameters):
         return self.extend(parameters)
 
-    def append(self, parameter):
+    def append(self, parameter, prefix=None):
         """Appends a given parameter at the end of the list.
 
         Arguments:
             parameter (nn.Parameter): parameter to append
         """
-        self.register_parameter(str(len(self)), parameter)
+        if isinstance(parameter, tuple):
+            k, parameter = parameter
+            param_key = _addPrefix(str(k), prefix)
+        else:
+            param_key = _addPrefix(str(len(self)), prefix)
+        self.register_parameter(param_key, parameter)
         return self
 
-    def extend(self, parameters):
-        """Appends parameters from a Python list at the end.
+    def extend(self, parameters, prefix=None):
+        """Appends parameters from a Python iterable to the end of the list.
 
         Arguments:
-            parameters (list): list of parameters to append
+            parameters (iterable): iterable of parameters to append
         """
-        if not isinstance(parameters, list):
-            raise TypeError("ParameterList.extend should be called with a "
-                            "list, but got " + type(parameters).__name__)
+        if not isinstance(parameters, Iterable) or isinstance(parameters, str):
+            raise TypeError("ParameterList.extend should be called with an "
+                            "iterable, but got " + type(parameters).__name__)
         offset = len(self)
         for i, param in enumerate(parameters):
-            self.register_parameter(str(offset + i), param)
+            if isinstance(param, tuple):
+                k, param = param
+                param_key = _addPrefix(str(k), prefix)
+            else:
+                param_key = _addPrefix(str(offset + i), prefix)
+            self.register_parameter(param_key, param)
         return self
+
+    def __repr__(self):
+        tmpstr = self.__class__.__name__ + '(\n'
+        for k, p in self._parameters.items():
+            size_str = 'x'.join(str(size) for size in p.size())
+            device_str = '' if not p.is_cuda else \
+                ' (GPU {})'.format(p.get_device())
+            parastr = '[Parameter ({}) of size {}{}]'.format(
+                torch.typename(p.data), size_str, device_str)
+            tmpstr = tmpstr + '  (' + k + '): ' + parastr + '\n'
```

Comment:
I'd rather not use the same formatting that we use to display submodules to show parameters. It's misleading. Let's just print it as a flat list of these param descriptions (without the names).

Reply:
Ok, do you mean like

    Parameter containing: torch.FloatTensor of size 5x5

or

    Parameter containing:
     0  0  0  0  0
     0  0  0  0  0
     0  0  0  0  0
     0  0  0  0  0
     0  0  0  0  0
    [torch.FloatTensor of size 5x5]

I am assuming something like the first, but I just want to make sure.  I couldn't find a function that prints out both the size and type of a parameter, without printing out the tensor data as well.